model-api-gen/src/main/kotlin/org/modelix/metamodel/generator/TypescriptMMGenerator.kt

```diff
@@ -56,274 +55,40 @@ class TypescriptMMGenerator(val outputDir: Path, val nameConfig: NameConfig = Na
 
     private fun generateRegistry(languages: ProcessedLanguageSet) {
         outputDir.resolve("index.ts").fixFormatAndWriteText(
-            """
-            import { LanguageRegistry } from "@modelix/ts-model-api";
-            ${languages.getLanguages().joinToString("\n") { """
-                import { ${it.simpleClassName()} } from "./${it.simpleClassName()}";
-            """
-            }}
-            export function registerLanguages() {
-                ${languages.getLanguages().joinToString("\n") { """
-                    LanguageRegistry.INSTANCE.register(${it.simpleClassName()}.INSTANCE);
-            """
-            }}
-            }
+            """$fileHeaderText
+
             """,
         )
     }
 
     private fun generateLanguage(language: ProcessedLanguage): String {
-        val conceptNamesList = language.getConcepts()
-            .joinToString(", ") { it.conceptWrapperInterfaceName() }
-
-        return """
+        return """$fileHeaderText
             import {
-                ChildListAccessor,
-                GeneratedConcept,
-                GeneratedLanguage,
-                IConceptJS,
-                INodeJS,
-                ITypedNode,
-                SingleChildAccessor,
-                TypedNode,
-                LanguageRegistry
-            } from "@modelix/ts-model-api";
-
-            ${language.languageDependencies().joinToString("\n") {
-            """import * as ${it.simpleClassName()} from "./${it.simpleClassName()}";"""
-        }}
-
-            export class ${language.simpleClassName()} extends GeneratedLanguage {
-                public static INSTANCE: ${language.simpleClassName()} = new ${language.simpleClassName()}();
-                constructor() {
-                    super("${language.name}")
-
-                    ${language.getConcepts().joinToString("\n") { concept ->
-            """
-                        this.nodeWrappers.set("${concept.uid}", (node: INodeJS) => new ${concept.nodeWrapperImplName()}(node))
-            """
-        }}
-                }
-                public getConcepts() {
-                    return [$conceptNamesList]
-                }
-            }
+                ${
+            listOf(
+                language.name.substringBefore("."),
+                "org",
+                "INodeJS",
+                "IConceptJS",
+            ).distinct().sorted().joinToString(",\n")
+        }
+            } from "$runtimeNpmPackage";
```

Comment:
That is weird, that one package (`$runtimeNpmPackage`/`@modelix/model-client`) must provide the access to the model client (server connnection, untyped intefaces) and the generated languages.

I thought of `@modelix/model-client` as a library that modelix would provide.
And the generated languages would come from another package and depend upon `@modelix/model-client`.